ansible/files/admin_api_scripts/pg_upgrade_scripts/initiate.sh

```diff
@@ -175,18 +187,50 @@ function initiate_upgrade {
     PGDATAOLD=$(cat "$POSTGRES_CONFIG_PATH" | grep data_directory | sed "s/data_directory = '\(.*\)'.*/\1/")
 
     PGDATANEW="$MOUNT_POINT/pgdata"
-    PG_UPGRADE_BIN_DIR="/tmp/pg_upgrade_bin/$PGVERSION"
-    PGBINNEW="$PG_UPGRADE_BIN_DIR/bin"
-    PGLIBNEW="$PG_UPGRADE_BIN_DIR/lib"
-    PGSHARENEW="$PG_UPGRADE_BIN_DIR/share"
 
     # running upgrade using at least 1 cpu core
     WORKERS=$(nproc | awk '{ print ($1 == 1 ? 1 : $1 - 1) }')
+
+    # To make nix-based upgrades work for testing, create a pg binaries tarball with the following contents:
+    #  - nix_flake_version - a7189a68ed4ea78c1e73991b5f271043636cf074
+    # Where the value is the commit hash of the nix flake that contains the binaries
+
+    if [ -n "$IS_LOCAL_UPGRADE" ]; then
+        mkdir -p "$PG_UPGRADE_BIN_DIR"
+        mkdir -p /tmp/persistent/
+        echo "a7189a68ed4ea78c1e73991b5f271043636cf074" > "$PG_UPGRADE_BIN_DIR/nix_flake_version"
+        tar -czf "/tmp/persistent/pg_upgrade_bin.tar.gz" -C "/tmp/pg_upgrade_bin" .
+        rm -rf /tmp/pg_upgrade_bin/
+    fi
```

Comment:
There's a real mix of using variables for directories and then hardcoding them in other places. I see potential for errors to creep in

Reply:
This is just for local testing, to aid in bypassing the need to run the full upgrade flow 😬